reader-osm/src/test/java/com/graphhopper/GraphHopperIT.java

```diff
@@ -1385,17 +1416,26 @@ public void testDisablingLM() {
 
     @Test
     public void testTurnCostsOnOff() {
+        final String profile1 = "profile1";
+        final String profile2 = "profile2";
+        final String vehicle = "car";
+        final String weighting = "fastest";
         GraphHopper hopper = createGraphHopper("car|turn_costs=true").
                 setOSMFile(MOSCOW).
+                setProfiles(
+                        new ProfileConfig(profile1).setVehicle(vehicle).setWeighting(weighting).setTurnCosts(false),
+                        new ProfileConfig(profile2).setVehicle(vehicle).setWeighting(weighting).setTurnCosts(true)
+                ).
                 setStoreOnFlush(true);
         hopper.importOrLoad();
 
-        // no edge_based parameter -> use edge-based (since encoder supports it and no CH)
-        assertMoscowEdgeBased(hopper, "none", false);
-        // edge_based=false -> use node-based
-        assertMoscowNodeBased(hopper, "false", false);
-        // edge_based=true -> use edge-based
-        assertMoscowEdgeBased(hopper, "true", false);
+        GHRequest req = new GHRequest(55.813357, 37.5958585, 55.811042, 37.594689);
+        // without turn costs
+        req.setProfile("profile1");
+        assertEquals(400, hopper.route(req).getBest().getDistance(), 1);
+        // with turn costs
+        req.setProfile("profile2");
+        assertEquals(1044, hopper.route(req).getBest().getDistance(), 1);
```

Comment:
> To improve readability also in tests we could remove the comments and reuse the previously defined variable carWithoutTurnCosts="profile1" instead of just "profile1"?

Yes. Good point.

> And why are the assertMoscowEdge/NodeBased calls removed? Because it is expected to work once the converter is tested?

They weren't removed, but only inlined, because it no longer made sense to re-use the code in this method